lab-brandon/MyArrayListTest.java

```diff
@@ -0,0 +1,158 @@
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class MyArrayListTest {
+
+    @org.junit.jupiter.api.Test
+    void add() {
+        System.out.println("---- Tests for the add() method ----");
+
+        MyArrayList arr = new MyArrayList();
+
+        int val = 36;
+
+        String result = (Arrays.toString(arr.add(val)));
+        String expected = Arrays.toString(new int[] {36, 0, 0, 0, 0, 0, 0, 0, 0, 0});
+        System.out.println("Expected: " + expected);
+        System.out.println("Result: " + result);
+
+        assertEquals(expected, result) ;
+    }
+
+    // Tests for the add at index method
+
+    @org.junit.jupiter.api.Test
+    void addAtIndex() {
+        System.out.println("---- Tests for the add() at index method ----");
+
+        MyArrayList arr = new MyArrayList();
+
+        int val = 36;
+        int index = 4;
+
+        String result = Arrays.toString(arr.addAtIndex(val, index));
+        String expected = Arrays.toString(new int[] {0, 0, 0, 0, 36, 0, 0, 0, 0, 0});
+        System.out.println("Expected: " + expected);
+        System.out.println("Result: " + result);
+        assertEquals(expected, result);
+    }
+
+    // Tests for the remove value method
+
+    @org.junit.jupiter.api.Test
+    void removeValue() {
+        System.out.println("---- Tests for the removeValue() method ----");
+
+        MyArrayList arr = new MyArrayList();
+        arr.add(24);
+        arr.add(15);
+        System.out.println("Input: " + Arrays.toString(arr.add(36)));
+
+        int val = 36;
+
+        String result = (Arrays.toString(arr.removeValue(val)));
+        String expected = Arrays.toString(new int[] {24, 15, 0, 0, 0, 0, 0, 0, 0, 0});
+        System.out.println("Expected: " + expected);
+        System.out.println("Result: " + result);
+        assertEquals(expected, result);
+    }
```

Comment:
Try adding a test for a case where a user tries to remove a value that is not in the ArrayList  :)  This will help answer the infinite-or-not while loop question